### ISSUE TITLE
[22.0] build: Version info ignores a tag

### ISSUE THREAD

**Issue by hebasto** [bitcoin/bitcoin#22623]
When building against a tagged commit (e.g., https://github.com/bitcoin/bitcoin/releases/tag/v22.0rc2) the version info should pull a tag rather a git commit hash.

Compare:
```
$ bin/bitcoind -version
Bitcoin Core version v0.21.0rc2
Copyright (C) 2009-2020 The Bitcoin Core developers

Please contribute if you find Bitcoin Core useful. Visit
<https://bitcoincore.org/> for further information about the software.
The source code is available from <https://github.com/bitcoin/bitcoin>.

This is experimental software.
Distributed under the MIT software license, see the accompanying file COPYING
or <https://opensource.org/licenses/MIT>

```

and

```
$ bin/bitcoind -version
Bitcoin Core version v22.0.0-g873fbc745d037ad43570f81e58334c397bc95477
```

**Comment by MarcoFalke** [comment]
Is this a bug? I assumed it was a desired change, but I might be mistaken.

**Comment by hebasto** [comment]
> I assumed it was a desired change, but I might be mistaken.

Maybe it's my mistake? Did I miss a discussion of that change?

**Comment by hebasto** [comment]
The reason of such behavior is https://github.com/bitcoin/bitcoin/blob/873fbc745d037ad43570f81e58334c397bc95477/share/genbuild.sh#L23 where `git rev-parse --is-inside-work-tree` does not return `true`.

**Comment by hebasto** [comment]
cc @dongcarl

**Comment by hebasto** [comment]
Fixed in #22625.

**Comment by achow101** [comment]
I have an alternative solution in #22680

**Comment by dongcarl** [comment]
Fix for the underlying problem in: #22685

**Comment by fanquake** [comment]
Fixed in #22685.
